Lodgify.Payments.Stripe.Server.UnitTests/WebhookControllerTests.cs

```diff
@@ -0,0 +1,117 @@
+using System.Text;
+using Lodgify.Payments.Stripe.Application.UseCases.UpdateAccount;
+using Lodgify.Payments.Stripe.Infrastructure.Settings;
+using Lodgify.Payments.Stripe.Server.Controllers.v1.External;
+using Lodgify.Payments.Stripe.Server.UnitTests.Payloads;
+using Lodgify.Payments.Stripe.Shared;
+using Lodgify.Payments.Stripe.Shared.Payloads;
+using MediatR;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using NSubstitute;
+
+namespace Lodgify.Payments.Stripe.Server.UnitTests;
+
+using Xunit;
+
+public class WebhookControllerTests
+{
+    private readonly ISender _mediatorSenderMock;
+    private readonly IOptions<StripeSettings> _stripeSettingsMock;
+    private readonly WebhooksController _controller;
+
+    public WebhookControllerTests()
+    {
+        _mediatorSenderMock = Substitute.For<ISender>();
+        _stripeSettingsMock = Substitute.For<IOptions<StripeSettings>>();
+        _stripeSettingsMock.Value.Returns(new StripeSettings { WebhookSecret = "test_webhook_secret" });
+        _controller = new WebhooksController(_mediatorSenderMock, _stripeSettingsMock, Substitute.For<ILogger<WebhooksController>>());
+    }
+
+    [Fact]
+    public async Task HandleEvent_ShouldReturnBadRequest_WhenStripeSignatureHeaderIsMissing()
+    {
+        // Arrange
+        var context = new DefaultHttpContext();
+        context.Request.Body = new MemoryStream(Encoding.UTF8.GetBytes("{}"));
+        _controller.ControllerContext = new ControllerContext
+        {
+            HttpContext = context
+        };
+
+        // Act
+        var result = await _controller.HandleEvent(CancellationToken.None);
+
+        // Assert
+        var badRequestResult = Assert.IsType<BadRequestObjectResult>(result);
+        Assert.Equal("Stripe-Signature Header is missing", badRequestResult.Value);
```

Comment:
How about using everywhere FluentAssertions?
```suggestion
        badRequestResult.Value.Should().Be("Stripe-Signature Header is missing");
```